src/main/webapp/WEB-INF/view/admin.jsp

```diff
@@ -37,10 +39,24 @@
     <h1>Admin Page</h1>
 
     <% if(request.getAttribute("error") != null){ %>
-        <h2 style="color:red"><%= request.getAttribute("error") %></h2>
-    <% } %>
 
-    <p>Hello, administrator!</p>
+      <h2 style="color:red"><%= request.getAttribute("error") %></h2>
+
+    <% } else { %>
+
+      <%
```

Comment:
It might be cleaner to do away with these variables any simply call request.getAttribute when you need it. We wouldn't have to import java.lang.Integer then either.

Reply:
Acknowledged. (will fix in next PR)